File: javascript/lasagna-master/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

export function cookingStatus(remainingTime) {
  switch (remainingTime) {
    case undefined:
      return "You forgot to set the timer.";
    case 0:
      return "Lasagna is done.";
    default:
      return "Not done, please wait.";
  }
}

export function preparationTime(numberOfLayers, timePerLayer = 2) {
  return numberOfLayers.length * timePerLayer;
}

export function quantities(layers) {
  const result = { noodles: 0, sauce: 0.0 };
  for (const layer of layers) {
    if (layer === "noodles") {
      result.noodles += 50;
    } else if (layer === "sauce") {
      result.sauce += 0.2;
    }
  }

  return result;
}

export function addSecretIngredient(friendsList, myList) {
  myList.push(friendsList[friendsList.length - 1]);
}

export function scaleRecipe(recipe, numberOfPortions) {
  let scaled = {};
  for (const ingredient in recipe) {
    scaled[ingredient] = (recipe[ingredient] / 2) * numberOfPortions;
  }

  return scaled;
}

File: go/lasagna-master/lasagna_master.go
package lasagna

// PreparationTime takes layers and the average time and returns the preparation time.
func PreparationTime(layers []string, averageTimePerLayer int) int {
	if averageTimePerLayer == 0 {
		averageTimePerLayer = 2
	}

	return len(layers) * averageTimePerLayer
}

// Quantities takes layers and returns the amount of noodles and sauce required.
func Quantities(layers []string) (noodles int, sauce float64) {
	for _, layer := range layers {
		if layer == "noodles" {
			noodles += 50
		} else if layer == "sauce" {
			sauce += 0.2
		}
	}

	return
}

// AddSecretIngredient adds the secret ingredient to my list.
func AddSecretIngredient(friendsList []string, myList []string) {
	myList[len(myList)-1] = friendsList[len(friendsList)-1]
}

// ScaleRecipe scales the recipe.
func ScaleRecipe(amounts []float64, scale int) []float64 {
	var scaled = make([]float64, len(amounts))

	for i, amount := range amounts {
		scaled[i] = amount * float64(scale) / 2
	}

	return scaled
}
